R/checkLinking.r

```diff
@@ -1,100 +1,160 @@
 ### checkLink() is called by defineModel()
 
-### prueft, ob Design verlinkt ist: checkLinking(design[1:15,c(1:5,ncol(design))], bookletColumn = "TH")
-checkLinking <- function(design, blocks=NULL, bookletColumn=NULL, verbose=FALSE) {
-      design<- eatTools::makeDataFrame(design)
-      if(!all(sapply(design, inherits, what="character"))) {design <- data.frame(lapply(design, as.character), stringsAsFactors=FALSE)}
-      design<- checkOrCreateBookletColumn(des = design, bc = bookletColumn)
-      desL  <- eatTools::facToChar(reshape2::melt(design[["design"]], id.vars = design[["book"]], na.rm=TRUE, variable.name="blockPos", value.name="blockName"))
-      if(!is.null(blocks)) {
-          stopifnot(is.vector(blocks))
-          if(!inherits(blocks, "character")) {stop("'block' must be a character vector.")}
-          stopifnot(length(intersect(unlist(design), blocks)) > 0)
-          notInDe<- setdiff(blocks,unique(desL[,"blockName"]))
-          if(length(notInDe)>0) {warning(paste0(length(notInDe), " elements of 'blocks' vector missing in design."))}
-          desL   <- desL[which(desL[,"blockName"] %in% blocks),]                ### geht einfacher zu selektieren
+### ----------------------------------------------------------------------------
+### checks whether design is linked. checkLinking(design[1:15,c(1:5,ncol(design))], bookletColumn = "TH")
+
+checkLinking <- function(design, blocks=NULL, bookletColumn=NULL, verbose=FALSE){
+  # checks
+  checkmate::assert_character(blocks, null.ok = TRUE)
+  checkmate::assert_vector(bookletColumn, null.ok = TRUE)

```

Comment:
I think this assertion is expendable as the argument `bookletColumns` is passed to the `eatTools::existsBackgroundVariables()` function, where the corresponding checks already take place. For reasons of economy, this line could be omitted.